### ISSUE TITLE
C++ standard library breaks in new emscripten builds..

### ISSUE THREAD

**Issue by terop2** [emscripten-core/emscripten#20282]
Uncaught RuntimeError: Aborted(segmentation fault)
    at abort (web_page_highmem.js?1695042879:555:40)
    at segfault (web_page_highmem.js?1695042879:288:2)
    at web_page_highmem.wasm.SAFE_HEAP_LOAD_i32_4_4 (web_page_highmem.wasm?1695042879:0x618c17)
    at web_page_highmem.wasm.std::__2::basic_ostream<char, std::__2::char_traits<char>>::sentry::~sentry() (web_page_highmem.wasm?1695042879:0x5f86e1)
    at web_page_highmem.wasm.std::__2::basic_ostream<char, std::__2::char_traits<char>>::operator<<(int) (web_page_highmem.wasm?1695042879:0x5f8b48)
    at web_page_highmem.wasm.add_params_linkage(std::__2::vector<CodeGenLine, std::__2::allocator<CodeGenLine>>&, std::__2::vector<CodeGenVectors, std::__2::allocator<CodeGenVectors>>&, bool&, std::__2::map<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char>>, int, std::__2::less<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char>>>, std::__2::allocator<std::__2::pair<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char>> const, int>>>) (web_page_highmem.wasm?1695042879:0x342e1)
    at web_page_highmem.wasm.GameApi::execute_codegen(GameApi::Env&, GameApi::EveryApi&, std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char>>, GameApi::ExecuteEnv&) (web_page_highmem.wasm?1695042879:0x35af8)
    at web_page_highmem.wasm.run_callback(void*) (web_page_highmem.wasm?1695042879:0x63ac8)
    at web_page_highmem.wasm.MainLoopSplitter_win32_and_emscripten::Iter() (web_page_highmem.wasm?1695042879:0x9631e)
    at web_page_highmem.wasm.splitter_iter2(void*) (web_page_highmem.wasm?1695042879:0x4a65b)

Either I have some memory issues in the code or the ostream::operator<<(int) is failing for sentry problems?

**Comment by sbc100** [comment]
Could try building with `-fsantize=address`?   

Also, if you have time, could figure out exactly which change broke this for you: https://emscripten.org/docs/contributing/developers_guide.html#bisecting

**Comment by terop2** [comment]
I think the emscripten build system somehow picks different version of stdlib if you use newer emscripten versions. I've been switching between latest and 1.3.30 since the newer versions are kinda broken. but yes, I'll do -fsanitize=address.

**Comment by sbc100** [comment]
We did upgrade the version libc++ back in 9bb02c018.  That change was only just released in 3.1.46.   

Make sure do a clean build with switch between emscripten versions (and when bisecting).   It could be the stale object files can causes issues between versions (it should be rare but it can happen).

**Comment by sbc100** [comment]
We also updated libc++ in 973a7182e which got released in 3.1.41

**Comment by terop2** [comment]
I think part of the problem was that there was drag&drop -> javascript -> c++ -> javascript -> screen path for filenames, and those drag&drop filenames contained some characters that emscripten::val couldn't handle/I had tons of problems with utf8 filenames. I fixed that problem by applying base64 to the filename before passing it forward. => I got my bug fixed, but the system might be slightly fiddly relative to how utf8 filenames are being handled (its bad enough that I can't use those normal filenames anywhere)...

**Comment by terop2** [comment]
no, the ostream I'm thinking about should be completely independent of the bad filenames. But its possible that std::stringstream changes its internal locale to something else than english when odd characters are encountered. I got like bad_casts crashes earlier when utf8 characters were passed to the emscripten::val (via Module.ccall 'string').

**Comment by terop2** [comment]
I have already working solution to this / base64 encoding the filenames. Of course it doesnt solve the d&d, utf8, Module.ccall, c++ iostream, etc problems with it, but that sounds bigger task...I think we can now ignore this error unless you want to make the character encoding system more robust....

**Comment by terop2** [comment]
(i tried 3.1.30, 3.1.41 but they both worked fine in current builds)

**Comment by terop2** [comment]
bonus feature to anyone who reads this: https://ssh.meshpage.org/view.php?id=641807662
